wasihttp/types.go

```diff
@@ -0,0 +1,187 @@
+package wasihttp
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/bytecodealliance/wasm-tools-go/cm"
+	"github.com/ydnar/wasi-http-go/internal/wasi/http/types"
+	"github.com/ydnar/wasi-http-go/internal/wasi/io/streams"
+)
+
+// authority returns the authority of the request from the [http.Request] Host or URL.Host.
+//
+// [http.Request]: https://cs.opensource.google/go/go/+/refs/tags/go1.23.2:src/net/http/request.go;l=240-243
+func authority(req *http.Request) string {
+	auth := ""
+	if req.Host == "" {
+		auth = req.URL.Host
+	} else {
+		auth = req.Host
+	}
+	return auth
+}
+
+// toHeaders convert the [http.Header] to a wasi-http [types.Fields].
+//
+// [http.Header]: https://pkg.go.dev/net/http#Header
+// [types.Fields]: https://github.com/WebAssembly/wasi-http/blob/v0.2.0/wit/types.wit#L215
+func toHeaders(headers http.Header) (fields types.Fields) {
```

Comment:
Perhaps rename to toFields?

Also: idiomatic Go style doesn't need a named single result, so the return statement is a bit more explicit.